crates/modelardb_manager/src/metadata.rs

```diff
@@ -13,188 +13,478 @@
+    /// that name does not exist, return [`DeltaTableError`].
+    pub async fn table_sql(&self, table_name: &str) -> Result<String, DeltaTableError> {
+        let batch = self
+            .metadata_delta_lake
+            .query_table(
+                "table_metadata",
+                &format!("SELECT sql FROM table_metadata WHERE table_name = '{table_name}'"),
+            )
+            .await?;
+
+        let table_sql = array!(batch, 0, StringArray);
+        if table_sql.is_empty() {
+            let batch = self
+                .metadata_delta_lake
+                .query_table(
+                    "model_table_metadata",
+                    &format!(
+                        "SELECT sql FROM model_table_metadata WHERE table_name = '{table_name}'"
+                    ),
+                )
+                .await?;
 
-        row.try_get("sql")
+            let model_table_sql = array!(batch, 0, StringArray);
+            if model_table_sql.is_empty() {
+                Err(DeltaTableError::Generic(format!(
+                    "No table with the name '{table_name}' exists."
+                )))
+            } else {
+                Ok(model_table_sql.value(0).to_owned())
+            }
+        } else {
+            Ok(table_sql.value(0).to_owned())
+        }
     }
 
     /// Retrieve all rows of `column` from both the table_metadata and model_table_metadata tables.
     /// If the column could not be retrieved, either because it does not exist or because it could
-    /// not be converted to a string, return [`sqlx::Error`].
-    pub async fn table_metadata_column(&self, column: &str) -> Result<Vec<String>, sqlx::Error> {
-        let mut values: Vec<String> = vec![];
-
+    /// not be converted to a string, return [`DeltaTableError`].
+    pub async fn table_metadata_column(
+        &self,
+        column: &str,
+    ) -> Result<Vec<String>, DeltaTableError> {
         // Retrieve the column from both tables containing table metadata.
-        let select_statement = format!(
-            "SELECT {column} FROM table_metadata UNION SELECT {column} FROM model_table_metadata",
+        let table_metadata_batch = self
+            .metadata_delta_lake
+            .query_table(
+                "table_metadata",
+                &format!("SELECT {column} FROM table_metadata"),
+            )
+            .await?;
+
+        let model_table_metadata_batch = self
+            .metadata_delta_lake
+            .query_table(
+                "model_table_metadata",
+                &format!("SELECT {column} FROM model_table_metadata"),
+            )
+            .await?;
+
+        let table_metadata_column = array!(table_metadata_batch, 0, StringArray);
+        let model_table_metadata_column = array!(model_table_metadata_batch, 0, StringArray);
+
+        Ok(table_metadata_column
+            .iter()
+            .chain(model_table_metadata_column.iter())
+            .map(|column_value| column_value.unwrap().to_string())
```

Comment:
Is `column_value` not a already a reference to a `string`? If so, maybe use `to_owned()` instead of `to_string()`.

Reply:
Fixed.